File: Data/models/TopSite.swift
/* This Source Code Form is subject to the terms of the Mozilla Public License, v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import UIKit
import CoreData
import Foundation
import Shared

class TopSite: NSManagedObject {
    @NSManaged var title: String?
    @NSManaged var url: String?
    @NSManaged var syncUUID: String?
    @NSManaged var order: Int16
    @NSManaged var notifications: Int16
    @NSManaged var createdOn: Date?
    @NSManaged var favicon: FaviconMO?
}

File: android/java/org/chromium/chrome/browser/ntp_background_images/model/TopSite.java
/* Copyright (c) 2020 The Brave Authors. All rights reserved.
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this file,
 * You can obtain one at https://mozilla.org/MPL/2.0/. */

package org.chromium.chrome.browser.ntp_background_images.model;

public class TopSite {
    private final String mName;
    private final String mDestinationUrl;
    private final String mBackgroundColor;
    private final String mImagePath;

    public TopSite(String name, String destinationUrl, String backgroundColor, String imagePath) {
        mName = name;
        mDestinationUrl = destinationUrl;
        mBackgroundColor = backgroundColor;
        mImagePath = imagePath;
    }

    public String getName() {
        return mName;
    }        

    public String getDestinationUrl() {
        return mDestinationUrl;
    }

    public String getBackgroundColor() {
        return mBackgroundColor;
    }

    public String getImagePath() {
        return mImagePath;
    }
}